06week/dodgeBall.js

```diff
@@ -0,0 +1,373 @@
+'use strict';
+
+// let assert = require('assert');
+
+//this is my starting list. 
+const arrOfPeople = [
+  {
+    id: 2,
+    name: "Charles Young",
+    age: 55,
+    skillSet: "welding",
+    placeBorn: "Omaha, Nebraska"
+  },
+  {
+    id: 3,
+    name: "Judy Twilight",
+    age: 35,
+    skillSet: "fishing",
+    placeBorn: "Louisville, Kentucky"
+  },
+  {
+    id: 4,
+    name: "Cynthia Doolittle",
+    age: 20,
+    skillSet: "tic tac toe",
+    placeBorn: "Pawnee, Texas"
+  },
+  {
+    id: 5,
+    name: "John Willouby",
+    age: 28,
+    skillSet: "pipe fitting",
+    placeBorn: "New York, New York"
+  },
+  {
+    id: 6,
+    name: "Stan Honest",
+    age: 20,
+    skillSet: "boom-a-rang throwing",
+    placeBorn: "Perth, Australia"
+  },
+  {
+    id: 7,
+    name: "Mia Watu",
+    age: 17,
+    skillSet: "acrobatics",
+    placeBorn: "Los Angeles, California"
+  },
+  {
+    id: 8,
+    name: "Walter Cole",
+    age: 32,
+    skillSet: "jump rope",
+    placeBorn: "New Orleans, Louisiana"
+  },
+]
+
+//code the will be repeated a lot
+
+//remember the name needs to be a 'string'
+const setAtt = (element,id,name) => {element.setAttribute(id,name)}
+const appending = (parent, child) => {parent.appendChild(child)}
+const byID = (id) => document.getElementById(id)
+
+//get from DOM
+const body = byID('body')
+const main = byID('main')
+const lopParent = byID('lopParent')
+  const listElement = byID('people')
+const playersDiv = byID('playerParent')
+  const dbUL = byID('players')
+const blueDiv = byID('bParent')
+  const blueUL = byID('blue')
+const redDiv = byID('rParent')
+  const redUL = byID('red')
+
+const listPeopleChoices = () => {
+  arrOfPeople.map(person => {
+    // console.log('this this map the whole object?',person) 
+    //yes, the whole object is mapped. 
+    const li = document.createElement("li")
+    const button = document.createElement("button")
+      // console.log('button', button)
+    button.innerText = "Make Player"
+    button.addEventListener('click', function(){makePlayer(person.id)} )
+      li.appendChild(button)
+      li.appendChild(document.createTextNode(person.name + " - " + person.skillSet))
+      setAtt(li,'id',`${person.id}`)
+        // console.log(li)
+      listElement.append(li)
+    })
+    console.log('listElement',listElement)
+  }
+
+  class Player {
+    //these characteristic will have to be added to the individuals before they get added to the list of players
+    constructor(canThrowBall, canDodgeBall, hasPaid, isHealthy, yearsExperience){
+      //these would be the default answers on all individuals 
+      this.canThrowBall = canThrowBall;
+      this.canDodgeBall = canDodgeBall; 
+      this.hasPaid = hasPaid;
+      this.isHealthy = isHealthy;
+      this.yearsExperience = 0;
+    }
+  }
+
+//feel like this should be an extended class, but had too many issues 
+class Teammates {
```

Comment:
Interesting, this is really a Team class not a Teammate class.
The original idea was that a player is the instance of the person before they are assigned to a team, and a teammate is the isntance of a person after they are assigned to a team.

What you have wold also work, but it would have been better if it included a list of players in the Team class.